pkg/services/alerting/ticker.go

```diff
@@ -1,76 +1,77 @@
 package alerting
 
 import (
+	"fmt"
 	"time"
 
 	"github.com/benbjohnson/clock"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
 // Ticker is a ticker to power the alerting scheduler. it's like a time.Ticker, except:
 // * it doesn't drop ticks for slow receivers, rather, it queues up.  so that callers are in control to instrument what's going on.
-// * it automatically ticks every second, which is the right thing in our current design
-// * it ticks on intervalSec marks or very shortly after. this provides a predictable load pattern
+// * it ticks on interval marks or very shortly after. this provides a predictable load pattern
 //   (this shouldn't cause too much load contention issues because the next steps in the pipeline just process at their own pace)
 // * the timestamps are used to mark "last datapoint to query for" and as such, are a configurable amount of seconds in the past
-// * because we want to allow:
-//   - a clean "resume where we left off" and "don't yield ticks we already did"
-//   - adjusting offset over time to compensate for storage backing up or getting fast and providing lower latency
-//   you specify a lastProcessed timestamp as well as an offset at creation, or runtime
 type Ticker struct {
-	C           chan time.Time
-	clock       clock.Clock
-	last        time.Time
-	offset      time.Duration
-	newOffset   chan time.Duration
-	intervalSec int64
-	paused      bool
+	C        chan time.Time
+	clock    clock.Clock
+	last     time.Time
+	interval time.Duration
 }
 
-// NewTicker returns a ticker that ticks on intervalSec marks or very shortly after, and never drops ticks
-func NewTicker(last time.Time, initialOffset time.Duration, c clock.Clock, intervalSec int64) *Ticker {
+var (
+	lastTick = promauto.NewGauge(prometheus.GaugeOpts{
+		Namespace: "grafana",
+		Subsystem: "alerting",
+		Name:      "ticker_last_consumed_tick",
+		Help:      "Timestamp of the last consumed tick",
+	})
+	nextTick = promauto.NewGauge(prometheus.GaugeOpts{
+		Namespace: "grafana",
+		Subsystem: "alerting",
+		Name:      "ticker_next_tick",
+		Help:      "Timestamp of the next tick",
+	})
+	interval = promauto.NewGauge(prometheus.GaugeOpts{
+		Namespace: "grafana",
+		Subsystem: "alerting",
+		Name:      "ticker_interval",
+		Help:      "Duration of the ticker interval",
+	})
+)
+
+// NewTicker returns a Ticker that ticks on interval marks (or very shortly after) starting at c.Now(), and never drops ticks. interval should not be negative or zero.
+func NewTicker(c clock.Clock, interval time.Duration) *Ticker {
+	if interval <= 0 {
+		panic(fmt.Errorf("non-positive interval [%v] is not allowed", interval))
```

Comment:
I would prefer that we returned an error here instead of panic?

Reply:
I think panic, in this case, makes sense. 
1. `time.Ticker` also panics in this case.
2. the input from timer does not come from user-input. It is hardcoded in all cases. So, error with creating the ticker will (and must) result in panic.